Export private mod, and use `cfg`'ed path instead of re-exported one

### Describe the bug

frb_generated.rs contains an exported function which is in a private mod named `time`

Functions in this private mod are exported by frb, but they are also using the "full path" for current target instead of using the dynamic set using `cfg` based on arch.



### Steps to reproduce

https://github.com/fzyzcjy/flutter_rust_bridge/compare/master...Slixe:flutter_rust_bridge:bug-cfg-export

1. Delete all generated rs files, and `cargo clean`
2. `flutter_rust_bridge_codegen generate`
3. `cargo build --target wasm32-unknown-unknown --no-default-features --lib`

Error also occurs with private mod `time` using `cargo build` but the second error about inner mod `native` is not shown.

### Logs

```shell
error[E0432]: unresolved import `flutter_rust_bridge::for_generated::Lifetimeable`
  --> src/frb_generated.rs:29:64
   |
29 | use flutter_rust_bridge::for_generated::{transform_result_dco, Lifetimeable, Lockable};
   |                                                                ^^^^^^^^^^^^ no `Lifetimeable` in `for_generated`

error[E0432]: unresolved import `flutter_rust_bridge::for_generated::Lifetimeable`
  --> src/frb_generated.web.rs:10:64
   |
10 | use flutter_rust_bridge::for_generated::{transform_result_dco, Lifetimeable, Lockable};
   |                                                                ^^^^^^^^^^^^ no `Lifetimeable` in `for_generated`
   |
   = help: consider importing this unresolved item through its public re-export instead:
           crate::frb_generated::Lifetimeable

error[E0433]: failed to resolve: could not find `native` in `time`
   --> src/frb_generated.rs:105:75
    |
105 |                     let output_ok = Result::<_, ()>::Ok(crate::api::time::native::hello_world())?;
    |                                                                           ^^^^^^ could not find `native` in `time`
    |
note: found an item that was configured out
   --> src/api/time/mod.rs:20:5
    |
20  | mod native;
    |     ^^^^^^

error[E0603]: module `time` is private
   --> src/frb_generated.rs:105:69
    |
105 |                     let output_ok = Result::<_, ()>::Ok(crate::api::time::native::hello_world())?;
    |                                                                     ^^^^ private module
    |
note: the module `time` is defined here
   --> src/api/mod.rs:2:1
    |
2   | mod time;
    | ^^^^^^^^^

```


### Expected behavior

- FRB should not export private mod
- FRB should not use the "full path" of current target when its generated, but use re-exported functions/structs for better compatibility.

### Generated binding code

_No response_

### OS

Debian 11 Linux

### Version of `flutter_rust_bridge_codegen`

2.0.0-dev.41

### Flutter info

```shell
[✓] Flutter (Channel beta, 3.23.0-0.1.pre, on Debian GNU/Linux 11 (bullseye) 5.10.0-29-amd64, locale fr_FR.UTF-8)
    • Flutter version 3.23.0-0.1.pre on channel beta at /home/slixe/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2feea7a407 (il y a 2 semaines), 2024-06-06 10:19:10 +0700
    • Engine revision bb10c54666
    • Dart version 3.5.0 (build 3.5.0-180.3.beta)
    • DevTools version 2.36.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/slixe/Android/Sdk
    • Platform android-34, build-tools 30.0.3
    • ANDROID_HOME = /home/slixe/Android/Sdk
    • ANDROID_SDK_ROOT = /home/slixe/Android/Sdk
    • Java binary at: /home/slixe/IDE/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Debian clang version 11.0.1-2
    • cmake version 3.18.4
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2022.2)
    • Android Studio at /home/slixe/IDE/android-studio
    • Flutter plugin version 73.0.2
    • Dart plugin version 222.4560
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.89.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.90.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Debian GNU/Linux 11 (bullseye) 5.10.0-29-amd64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 125.0.6422.76

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```


### Version of `clang++`

Debian clang version 11.0.1-2

### Additional context

_No response_

Hum, I think `/// frb:ignore` on my `mod time;` is broken.
```rust
/// frb:ignore
mod time;
```

Full changes on your dart_minimal:
https://github.com/fzyzcjy/flutter_rust_bridge/compare/master...Slixe:flutter_rust_bridge:frb-ignore

What am I doing wrong ?

```
error[E0603]: module `time` is private
   --> src/frb_generated.rs:143:57
    |
143 |                         Result::<_, ()>::Ok(crate::api::time::native::get_current_time_millis())?;
    |                                                         ^^^^          ----------------------- function `get_current_time_millis` is not publicly re-exported
    |                                                         |
    |                                                         private module
    |
note: the module `time` is defined here
   --> src/api/mod.rs:3:1
    |
3   | mod time;
    | ^^^^^^^^^
```

>Feel free to PR! (I seemed to be too tired and forgot to say this before...)

I would love to, but you're doing too much advanced things for me :)

--- a/website/docs/guides/custom/attributes.md
+++ b/website/docs/guides/custom/attributes.md
@@ -6,7 +6,7 @@ For example, to make a function synchronous on Dart side, we can put `#[frb(sync
 ## Specify attributes as comments
 
 Not only can we write down `#[frb(something)]`,
-but we can also write comments `/// frb:something` to do the same thing.
+but we can also write comments `/// flutter_rust_bridge:something` to do the same thing.
 
 The latter is especially useful when the former cannot be used,
 such as when the target is a `mod`, or when the crate does not have dependency on `flutter_rust_bridge`.
@@ -19,7 +19,7 @@ But this can be easily spotted since it will have compile-time messages.
 ### Example
 
 ```rust
-/// frb:ignore
+/// flutter_rust_bridge:ignore
 pub fn f() {}
 ```
 

--- a/website/docs/guides/misc-features/ignoring.md
+++ b/website/docs/guides/misc-features/ignoring.md
@@ -1,6 +1,6 @@
 # Ignoring things
 
-The `#[frb(ignore)]` (or equivalently, `/// frb:ignore`) can be put on things to ignore them.
+The `#[frb(ignore)]` (or equivalently, `/// flutter_rust_bridge:ignore`) can be put on things to ignore them.
 
 To ignore a `mod` (module), we can only use the latter syntax because Rust grammar does not support the former yet.
 
@@ -15,6 +15,6 @@ pub struct ThisStructWillBeIgnore;
 #[frb(ignore)]
 pub fn this_function_will_be_ignored() {}
 
-/// frb:ignore
+/// flutter_rust_bridge:ignore
 mod this_submodule_will_be_ignored;
 ```

